homestar-runtime/src/event_handler.rs

```diff
@@ -54,11 +57,16 @@ pub(crate) struct EventHandler<DB: Database> {
     p2p_provider_timeout: Duration,
     db: DB,
     swarm: Swarm<ComposedBehaviour>,
+    cache: Arc<Cache<String, CacheValue>>,
     sender: Arc<channel::AsyncBoundedChannelSender<Event>>,
     receiver: channel::AsyncBoundedChannelReceiver<Event>,
     query_senders: FnvHashMap<QueryId, (RequestResponseKey, Option<P2PSender>)>,
     connected_peers: FnvHashMap<PeerId, ConnectedPoint>,
+    connected_peers_limit: u32,
+    discovered_peers: FnvHashMap<PeerId, PeerDiscoveryInfo>,
     request_response_senders: FnvHashMap<RequestId, (RequestResponseKey, P2PSender)>,
+    rendezvous_registration_ttl: Duration,
```

Comment:
@bgins thinking maybe we do have some substruct here: rendezvous, which has ttl and discovery interval and cookies? Trying to clean up this bit on length here. We may want the same for the two connected peers.

Reply:
Done! Also added `discovered_peers` to the rendezvous one because it is not relevant for peers discovered through mDNS. We may revisit if we use the proposed ambient discovery protocol in the future.

Was slightly unsure about the naming for the connected peers one. Went with this struct:

https://github.com/ipvm-wg/homestar/blob/1d4079b19884b9b130ac9a6274b96ed04f54720f/homestar-runtime/src/event_handler.rs#L104-L108

Which on the event handler is:

https://github.com/ipvm-wg/homestar/blob/1d4079b19884b9b130ac9a6274b96ed04f54720f/homestar-runtime/src/event_handler.rs#L64

And when used we have `event_handler.connections.peers` and `event_handler.connections.max_peers`. In the first one, having `connections` and `peers` seems a bit redundant? I might be overthinking it.